File: ListObjC/ListObjC/ViewControllers/TableViewController.m
//
//  TableViewController.m
//  ListObjC
//
//  Created by pablo borquez on 26/08/23.
//

#import "TableViewController.h"
#import "TableViewFactory.h"

@interface TableViewController ()

@property (nonatomic, strong) TableViewFactory *factory;

@end

@implementation TableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self prepareFactory];
    [self.navigationItem setTitle:@"Objective-C List"];
    [self.navigationItem setLargeTitleDisplayMode:UINavigationItemLargeTitleDisplayModeAlways];
}

- (void)prepareFactory {
    self.factory = [[TableViewFactory alloc] init];
    [self.factory prepareFactory:self.tableView];
}

@end

File: ListSwift/ListSwift/ViewControllers/TableViewController.swift
//
//  TableViewController.swift
//  ListSwift
//
//  Created by pablo borquez on 08/09/23.
//

import UIKit

class TableViewController: UITableViewController {
    private var factory: TableViewFactory?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.prepareFactory()
        self.navigationItem.title = "Swift List"
        self.navigationItem.largeTitleDisplayMode = .always
    }
    
    // MARK: - private
    
    private func prepareFactory() {
        self.factory = TableViewFactory()
        self.factory?.prepare(tableView: self.tableView)
    }
}
